--- a/src/synaptics.c	2012-09-12 18:28:09.000000000 +0400
+++ b/src/synaptics.c	2012-11-19 17:25:12.678030888 +0400
@@ -578,9 +578,9 @@
     emulateTwoFingerMinW = priv->minw + range * (7.0 / 16);
 
     /* Enable tap if we don't have a phys left button */
-    tapButton1 = priv->has_left ? 0 : 1;
-    tapButton2 = priv->has_left ? 0 : 3;
-    tapButton3 = priv->has_left ? 0 : 2;
+    tapButton1 = 1; /*priv->has_left ? 0 : 1;*/
+    tapButton2 = 3; /*priv->has_left ? 0 : 3;*/
+    tapButton3 = 2; /*priv->has_left ? 0 : 2;*/
 
     /* Enable multifinger-click if only have one physical button,
        otherwise clickFinger is always button 1. */
@@ -589,11 +589,11 @@
     clickFinger3 = (priv->has_right || priv->has_middle) ? 1 : 2;
 
     /* Enable vert edge scroll if we can't detect doubletap */
-    vertEdgeScroll = priv->has_double ? FALSE : TRUE;
+    vertEdgeScroll = TRUE; /*priv->has_double ? FALSE : TRUE;*/
     horizEdgeScroll = FALSE;
 
     /* Enable twofinger scroll if we can detect doubletap */
-    vertTwoFingerScroll = priv->has_double ? TRUE : FALSE;
+    vertTwoFingerScroll = FALSE; /*priv->has_double ? TRUE : FALSE;*/
     horizTwoFingerScroll = FALSE;
 
     /* Use resolution reported by hardware if available */
